core/src/main/java/org/elasticsearch/indices/IndicesService.java

```diff
@@ -430,7 +417,7 @@ private void removeIndex(String index, String reason, boolean delete) {
                 deleteIndexStore(reason, indexService.index(), indexSettings, false);
             }
         } catch (IOException ex) {
-            throw new ElasticsearchException("failed to remove index " + index, ex);
+            throw new ElasticsearchException("failed to remove index " + indexName, ex);
```

Comment:
I think I'd prefer `"failed to remove index " + index` here and get the toString for `Index`.